src/nft/components/view/NFTVideo.tsx

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useRef } from 'react';
+import { defaultNFTSvg } from '../../../internal/svg/defaultNFTSvg';
+import { cn } from '../../../styles/theme';
+import type { NFTError } from '../../types';
+import { useNFTContext } from '../NFTProvider';
+
+type NFTVideoReact = {
+  className?: string;
+  onLoading?: (mediaUrl: string) => void;
+  onLoaded?: () => void;
+  onError?: (error: NFTError) => void;
+};
+
+export function NFTVideo({
+  className,
+  onLoading,
+  onLoaded,
+  onError,
+}: NFTVideoReact) {
+  const { animationUrl, imageUrl } = useNFTContext();
+  const videoRef = useRef<HTMLVideoElement>(null);
+
+  useEffect(() => {
+    if (animationUrl && videoRef?.current) {
+      videoRef.current.onloadstart = () => {
+        onLoading?.(animationUrl);
+      };
+
+      videoRef.current.onloadeddata = () => {
+        onLoaded?.();
+      };
+
+      videoRef.current.onerror = (error: string | Event) => {
+        onError?.({
+          error: typeof error === 'string' ? error : error.type,
+          code: 'NmNVc01', // NFT module NFTVideo component 01 error
```

Comment:
Should we add all of the possible error codes in a error / const file and reference from it?

Reply:
I like this idea

Reply:
let me revisit that at the end